### DISCUSSION TITLE
EasyWechat 6 怎么解密微信消息推送

### DISCUSSION THREAD

**Discussion by qianshi-tech** [w7corp/easywechat#2750] [Q&A]
消息推送
https://developers.weixin.qq.com/miniprogram/dev/framework/server-ability/message-push.html
webman中检验signature的示例代码：
```php
$symfony_request = new SymfonyRequest($request->get(), $request->post(), [], $request->cookie(), [], [], $request->rawBody());
$symfony_request->headers = new HeaderBag($request->header());
$this->app->setRequestFromSymfonyRequest($symfony_request);
$server = $this->app->getServer();
$data = $server->getDecryptedMessage();
var_dump($data->Encrypt,$data->toArray()); //这个怎么解密
$response = $server->serve();
return $response->getBody();
```
检验后可以怎么解密吗？求助， 提前感谢您

```
$data->Encrypt;
```

**Comment by tmkdahko** [comment]
getDecryptedMessage就能拿到解密码后的数据，你在微信那边设置消息推送格式为xml应该就可以了，json格式的时候框架里有个bug，他把解密出来的消息直接当成xml格式去转数组，配置为json的时候转出来就是空数组所以拿不到觖密后的

**Reply by happynnan** [reply]
> getDecryptedMessage就能拿到解密码后的数据，你在微信那边设置消息推送格式为xml应该就可以了，json格式的时候框架里有个bug，他把解密出来的消息直接当成xml格式去转数组，配置为json的时候转出来就是空数组所以拿不到觖密后的

高级
